src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -904,19 +958,232 @@ static void _flush_log() {
 
 // Function for shared library JavaVM to exit HotSpot on a fatal error
 static void _fatal() {
+  Thread* thread = Thread::current_or_null_safe();
+  if (thread != NULL && thread->is_Java_thread()) {
+    JavaThread* jthread = (JavaThread*) thread;
+    JVMCIRuntime* runtime = jthread->libjvmci_runtime();
+    if (runtime != NULL) {
+      int javaVM_id = runtime->get_shared_library_javavm_id();
+      fatal("Fatal error in JVMCI shared library JavaVM[%d] owned by JVMCI runtime %d", javaVM_id, runtime->id());
+    }
+  }
   intx current_thread_id = os::current_thread_id();
   fatal("thread " INTX_FORMAT ": Fatal error in JVMCI shared library", current_thread_id);
 }
 
-JNIEnv* JVMCIRuntime::init_shared_library_javavm() {
-  JavaVM* javaVM = (JavaVM*) _shared_library_javavm;
-  if (javaVM == NULL) {
+JVMCIRuntime::JVMCIRuntime(JVMCIRuntime* next, int id, bool for_compile_broker) {
+  _init_state = uninitialized;
+  _shared_library_javavm = NULL;
```

Comment:
Thanks for the reminder. I'm tempted to now change `NULL` to `nullptr` in all `jvmci/*` files. What's the policy/recommendation on that? Don't bother/separate PR/piggy-back on this PR?

Reply:
I think most folks have been taking a nullptr in new code approach, without anyone doing substantial conversions.  Though Leo did look at a wholesale conversion at one point (it turned out to be a little premature, since we didn't really have nullptr yet).  There are some places where there are strange overload games being played because NULL is an unadorned `0` on some platforms, and I think Leo found some places where the selected overload was even wrong because of that.  I certainly wouldn't a wholesale conversion of JVMCI as part of this PR.

Reply:
Ok, I'll take this opportunity to convert some JVMCI code to `nullptr` and do more in subsequent PRs.